modules/apps/frontend-js/frontend-js-spa-web/test/app/App.js

```diff
@@ -125,64 +128,72 @@ describe('App', function () {
 	it('does not allow navigation for urls with hashbang when navigating to same basepath', () => {
 		this.app = new App();
 		this.app.addRoutes(new Route('/path', Screen));
-		globals.window = {
-			location: {
-				hash: '',
-				host: '',
-				hostname: '',
-				pathname: '/path',
-				search: '',
-			},
-		};
+
+		const restoreWindowLocation = mockWindowLocation({
+			hash: '',
+			host: '',
+			hostname: '',
+			pathname: '/path',
+			search: '',
+		});
+
 		expect(this.app.canNavigate('/path#hashbang')).toBe(false);
+
+		restoreWindowLocation();
```

Comment:
If any test fails, `restoreWindowLocation()` won't run. Can we save in `beforeEach()` and restore in `afterEach()` instead?